apps/judicial-system/web/src/components/Inputs/InputAdvocate.tsx

```diff
@@ -80,34 +118,64 @@ const DefenderInput: FC<Props> = ({
   )
 
   const handleLawyerChange = useCallback(
-    (selectedOption: SingleValue<ReactSelectOption>) => {
+    (
+      selectedOption: SingleValue<ReactSelectOption>,
+      isCivilClaim: boolean,
+      clientId?: string | null,
+    ) => {
       let updatedLawyer = {
         defenderName: '',
         defenderNationalId: '',
         defenderEmail: '',
         defenderPhoneNumber: '',
       }
 
+      let updatedSpokesperson = {
```

Comment:
I understand what you were thinking with this component, but it feels very complicated and confusing to read because of all the different handlings of civilClaimants vs. defendants and the 3 different types of advocates. 
I can see why you wanted to refactor and create a separate table for advocate info instead of storing it in 3 different places, all this logic is giving me a slight headache 😅
I don't think we should necessarily change it at this point though, if it works let's just get it out and keep in mind we'd like to simplify advocate handling overall from top to bottom when we have time.

Reply:
My suggestion is not to combine all the different types of advocates in a new table, but rather to inject relevant methods into the lawyer lookup component. Creating a new table would in my view combine unrelated entities just because the all come from the lawyer api. Too me, that sounds similar to creating a person table to store all human actors in the judicial system, i.e. defendants, prosecutors, judges, defenders, spokespersons, etc, just because they all have a national id.

Reply:
Yes, good point. I think we should still reconsider how we handle defender information as we have fields for that both in the case table and defendants table, which seems redundant. The civilClaimants spokesperson info is unrelated enough so that could be kept as is.

Reply:
Absolutely, we have been talking about that for some time. Recently, we added the possibility to have defenders on all R-Cases and a logical next step would be to handle defenders the same way in both R- and S-Cases.